File: examples/java/sample.java
// <r:package>
package com.example.demo;

// <r:imports>
import java.util.*;
import com.fasterxml.jackson.annotation.JsonProperty;

// <r:class>
public class UserManager {
    // <r:fields>
    private Map<Integer, User> users;
    private static final String DEFAULT_EMAIL = "noemail@example.com";
    
    // <r:constructor>
    public UserManager() {
        this.users = new HashMap<>();
    }
    
    // <r:methods>
    public void addUser(User user) {
        users.put(user.getId(), user);
    }
    
    public Optional<User> getUser(int id) {
        return Optional.ofNullable(users.get(id));
    }
    
    public List<User> getAllUsers() {
        return new ArrayList<>(users.values());
    }
    
    // <r:inner-class>
    public static class User {
        @JsonProperty("id")
        private int id;
        
        @JsonProperty("name")
        private String name;
        
        @JsonProperty("email")
        private String email;
        
        public User(int id, String name, String email) {
            this.id = id;
            this.name = name;
            this.email = email != null ? email : DEFAULT_EMAIL;
        }
        
        // Getters
        public int getId() { return id; }
        public String getName() { return name; }
        public String getEmail() { return email; }
    }
}

File: examples/go/sample.go
// <r:package>
package main

// <r:imports>
import (
	"fmt"
	"encoding/json"
	"net/http"
)

// <r:types>
type User struct {
	ID    int    `json:"id"`
	Name  string `json:"name"`
	Email string `json:"email"`
}

type UserService struct {
	users map[int]User
}

// <r:methods>
func NewUserService() *UserService {
	return &UserService{
		users: make(map[int]User),
	}
}

func (us *UserService) AddUser(user User) {
	us.users[user.ID] = user
}

func (us *UserService) GetUser(id int) (User, bool) {
	user, exists := us.users[id]
	return user, exists
}

// <r:handlers>
func (us *UserService) HandleGetUser(w http.ResponseWriter, r *http.Request) {
	// Implementation here
	fmt.Fprintf(w, "User handler")
}

// <r:main>
func main() {
	service := NewUserService()
	service.AddUser(User{ID: 1, Name: "Alice", Email: "alice@example.com"})
	
	user, exists := service.GetUser(1)
	if exists {
		fmt.Printf("Found user: %+v\n", user)
	}
}